File: Gabriel's examples/GabrielFedorowicz_Example1.ts
// Control Structure Example: Conditional Logic (TypeScript)
let x = 0;

// JavaScript *does* allow truthy/falsy coercion, but we avoid that here so the logic exactly matches the Dart version.
// Using === ensures a strict equality check without type coercion.
if (x === 0) {
  console.log("Value is zero");
} else {
  console.log("Value is not zero");
}
console.log("ts-node is working!");
// In TS/JS, you *could* write:
// if (!x) { ... }
// Since that would rely on truthy/falsy coercion, a feature not seen in Dart.

File: Gabriel's examples/GabrielFedorowicz_Example1.dart
// Control Structure Example: Conditional Logic (Dart)
void main() {
  int x = 0;

  // Dart does NOT allow truthy/falsy values (Only 0 or 1).
  // Conditions must evaluate to a strict bool.
  // Therefore we MUST explicitly compare x to 0.
  if (x == 0) {
    print("Value is zero");
  } else {
    print("Value is not zero");
  }

  // If you tried:
  // if (x) {}
  // Dart would throw a compile-time error.
}